packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -178,6 +178,18 @@ class RequestWrapper {
   async fetch({request}) {
     assert.atLeastOne({request});
 
+    if (this.behaviorCallbacks.requestWillFetch) {
+      for (let callback of this.behaviorCallbacks.requestWillFetch) {
+        const returnedRequest = callback({request});
+        if (returnedRequest && returnedRequest.then) {
```

Comment:
I'd rather we just stick to only supporting synchronous callbacks initially, unless we have an actual need for `async` callbacks or callbacks that return `Promise`s. It will make the documentation cleaner and simpler.